astropy/convolution/convolve.py

```diff
@@ -138,7 +138,7 @@ def _copy_input_if_needed(input, dtype=float, order='C', nan_treatment=None,
             # The advantage of `subok=True` is that it won't copy when array is an ndarray subclass. If it
             # is and `subok=False` (default), then it will copy even if `copy=False`. This uses less memory
             # when ndarray subclasses are passed in.
-            output = np.array(input, dtype=dtype, copy=False, order=order, subok=True)
+            output = npa.array(input, dtype=dtype, copy=False, order=order, subok=True)
```

Comment:
```suggestion
            if npa == np:
                output = np.array(input, dtype=dtype, copy=False, order=order, subok=True)
            else:
                output = npa.array(input, dtype=dtype)
```
Dask `array` does not support the other kwargs, so this raises when passing `npa=dask.array` through.
The other cases would also need to be taken care off, though I don't have a good idea how to deal with the `copy=True` cases for dask.